crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -14,7 +14,72 @@
 # limitations under the License.
 #
 
+require "open3"
+
 module Api
-  class Crowbar
+  class Crowbar < Tableless
+    attr_reader :version
+
+    def initialize
+      @version = ENV["CROWBAR_VERSION"]
+    end
+
+    def status
+      {
+        version: @version
+      }
+    end
+
+    def upgrade
+      status.merge!(
+        upgrade: {
+          upgrading: upgrading,
+          success: success,
+          failed: failed
+        }
+      )
+    end
+
+    def upgrade!
+      upgrade_script = Rails.root.join("..", "bin", "upgrade_admin_server.sh")
+
+      if upgrade_script.exist?
+        pid = spawn("sudo #{upgrade_script}")
+        Process.detach(pid)
+        Rails.logger.info("#{upgrade_script} executed with pid: #{pid}")
+
+        true
+      else
+        msg = "Could not find #{upgrade_script}"
+        Rails.logger.error(msg)
+        errors.add(:upgrade, msg)
+
+        false
+      end
+    end
+
+    def maintenance_updates_installed?
```

Comment:
now that you say, that was not fully correct indeed. I guess we just need to check for those two:

```
       100 - ZYPPER_EXIT_INF_UPDATE_NEEDED
           Returned by the patch-check command if there are patches available for installation.

       101 - ZYPPER_EXIT_INF_SEC_UPDATE_NEEDED
           Returned by the patch-check command if there are security patches available for
           installation.
```

do you have anything else in mind what we need to check for?